app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt

```diff
@@ -0,0 +1,491 @@
+                avatarProgressBar.visibility = View.VISIBLE
+                avatarPreview.visibility = View.INVISIBLE
+                avatarButton.setImageDrawable(null)
+
+            }
+            EditProfileActivity.PickType.HEADER -> {
+                headerProgressBar.visibility = View.VISIBLE
+                headerPreview.visibility = View.INVISIBLE
+                headerButton.setImageDrawable(null)
+            }
+        }
+    }
+
+    private fun endMediaPicking() {
+        avatarProgressBar.visibility = View.GONE
+        headerProgressBar.visibility = View.GONE
+
+        currentlyPicking = PickType.NOTHING
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        when (requestCode) {
+            AVATAR_PICK_RESULT -> {
+                if (resultCode == Activity.RESULT_OK && data != null) {
+                    CropImage.activity(data.data)
+                            .setInitialCropWindowPaddingRatio(0f)
+                            .setAspectRatio(AVATAR_SIZE, AVATAR_SIZE)
+                            .start(this)
+                } else {
+                    endMediaPicking()
+                }
+            }
+            HEADER_PICK_RESULT -> {
+                if (resultCode == Activity.RESULT_OK && data != null) {
+                    CropImage.activity(data.data)
+                            .setInitialCropWindowPaddingRatio(0f)
+                            .setAspectRatio(HEADER_WIDTH, HEADER_HEIGHT)
+                            .start(this)
+                } else {
+                    endMediaPicking()
+                }
+            }
+            CropImage.CROP_IMAGE_ACTIVITY_REQUEST_CODE -> {
+                val result = CropImage.getActivityResult(data)
+                when (resultCode) {
+                    Activity.RESULT_OK -> beginResize(result.uri)
+                    CropImage.CROP_IMAGE_ACTIVITY_RESULT_ERROR_CODE -> onResizeFailure()
+                    else -> endMediaPicking()
+                }
+            }
+        }
+    }
+
+    private fun beginResize(uri: Uri) {
+        beginMediaPicking()
+        val width: Int
+        val height: Int
+        val cacheFile: File
+        when (currentlyPicking) {
+            EditProfileActivity.PickType.AVATAR -> {
+                width = AVATAR_SIZE
+                height = AVATAR_SIZE
+                cacheFile = getCacheFileForName(AVATAR_FILE_NAME)
+            }
+            EditProfileActivity.PickType.HEADER -> {
+                width = HEADER_WIDTH
+                height = HEADER_HEIGHT
+                cacheFile = getCacheFileForName(HEADER_FILE_NAME)
+            }
+            else -> {
+                throw AssertionError("PickType not set.")
+            }
+        }
+        ResizeImageTask(contentResolver, width, height, cacheFile, object : ResizeImageTask.Listener {
```

Comment:
Of course it's not perfect that it captures activity but it shouldn't be too slow so I'm not sure if it's worth doing something with it.

Reply:
I am pretty sure that this AsyncTask does not have a reference to the activity. Thats the reason why the function below is outside the class, the function is not accessible if it is a member of the class (it doesnt compile).